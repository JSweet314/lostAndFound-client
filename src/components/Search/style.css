.search-form {
  align-items: center;
  display: flex;
  justify-content: center;
  width: 100%;
}

.search-form__search {
  padding: 0.4rem 1rem; 
  width: 40%;
  font-size: 3rem;
  border: 2px solid #625049;
  border-right: none;
  color: #625049;
}

.search-form__search::placeholder {
  font-family: 'Exo', sans-serif;
}

.search-form__submit {
  background-color: #EBC9AD;
  border: 2px solid #625049;
  border-left: none;
  font-size: 3rem;
  font-family: 'Exo', sans-serif;
}

.search-form__submit:hover {
  background-color: #625049;
  color: #EBC9AD;
}