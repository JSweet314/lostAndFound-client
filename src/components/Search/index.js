import React, { Component } from 'react';
import './style.css';

export default class Search extends Component {
  constructor() {
    super();
    this.state = {
      searchValue: ''
    };
  }

  handleOnChange = event => {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  }

  render() {
    const { searchValue } = this.state;
    const isDisabled = searchValue ? false : true;
    return (
      <form className='search-form'>
        <input
          type='text'
          className='search-form__search'
          value={searchValue}
          name='searchValue'
          onChange={event => this.handleOnChange(event)}
          aria-label='item search'
          placeholder="What are we lookin' for?" />
        <button 
          className='search-form__submit'
          disabled={isDisabled}
          type='submit'>
          Search
        </button>
      </form>
    );
  }
}