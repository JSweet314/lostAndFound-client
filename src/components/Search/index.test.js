import React from 'react';
import {shallow} from 'enzyme';
import Search from './index';

describe('Search', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Search />);
  });

  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should store a changing searchValue in local state', () => {
    wrapper.instance().handleOnChange({target: {
      name: 'searchValue',
      value: 'iPhone'
    }});
    expect(wrapper.state('searchValue')).toEqual('iPhone');
  });
});