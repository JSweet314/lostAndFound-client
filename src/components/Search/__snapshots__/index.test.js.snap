// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          aria-label="item search"
          className="search-form__search"
          name="searchValue"
          onChange={[Function]}
          placeholder="What are we lookin' for?"
          type="text"
          value=""
/>,
        <button
          className="search-form__submit"
          type="submit"
>
          Search
</button>,
      ],
      "className": "search-form",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-label": "item search",
          "className": "search-form__search",
          "name": "searchValue",
          "onChange": [Function],
          "placeholder": "What are we lookin' for?",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Search",
          "className": "search-form__submit",
          "type": "submit",
        },
        "ref": null,
        "rendered": "Search",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            aria-label="item search"
            className="search-form__search"
            name="searchValue"
            onChange={[Function]}
            placeholder="What are we lookin' for?"
            type="text"
            value=""
/>,
          <button
            className="search-form__submit"
            type="submit"
>
            Search
</button>,
        ],
        "className": "search-form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "item search",
            "className": "search-form__search",
            "name": "searchValue",
            "onChange": [Function],
            "placeholder": "What are we lookin' for?",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search",
            "className": "search-form__submit",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Search",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
